// cypress/integration/PageObjects/loginPage.js
import BasePage from 'cypress/integration/PageObjects/basePage'


/// <reference types="cypress-xpath" />


export default class LoginPage extends BasePage {

    constructor() {
        super()
    }

    
    LoginBtn = "[class='_1_3w1N']";
    username     = "[class='_2IX_2- VJZDxU']";
    password     = "[class='_2IX_2- _3mctLh VJZDxU']";
    submitButton = "[class='_2KpZ6l _2HKlqd _3AWRsL']";
    Mobiles = ".eFQ30H:nth-child(3) .xtXmba";
    SearchBox = ".\_3704LK";
    SelectData = ".\_3YgSsQ:nth-child(1) .\_1aPR5_";


    clickOnSearchBox() {
        BasePage.pause(60);
        const searchbox = cy.get(this.SearchBox);
        searchbox.click({force: true});
    }


    productSelection() {
        BasePage.pause(10000);
        cy.get('div').contains('realme C11 2021 (Cool Blue, 32 GB)').click({force: true});
        BasePage.pause(1000);

        cy.window();

        cy.get('._36yFo0').focus().type('500019');
        cy.get('._36yFo0').type('500019') ;
        cy.get('._2P_LDn').click({force: true});
        cy.go('back');
        
    }



    clickOnSelectData() {
        BasePage.pause(60);
        const button = cy.get(this.SelectData);
        button.click({force: true});
        BasePage.pause(100);
    }
    getSearchBox(credType) {
      //  BasePage.pause(60);
        let field = cy.get(this.SearchBox);
        field.clear();
        field.type(credType);
    }

    clickOnMobilesBtnn() {
        BasePage.pause(100);
        const button = cy.get(this.Mobiles);
        button.click({force: true});
    }

    clickOnLoginBtnn() {
        const button = cy.get(this.LoginBtn);
        button.click({force: true});
        BasePage.pause(100);
    }

    getUserName(credType) {
        let field = cy.get(this.username);
        field.clear();
        field.type(credType);
    }

    getPassword(credType) {
        let field = cy.get(this.password);
        field.clear();
        field.type(credType);
    }

    clickOnLoginButton() {
        const button = cy.get(this.submitButton);
        button.click();
    }

}
